src/Libraries/CoreNodes/Sorting.cs

```diff
@@ -55,10 +55,18 @@ public static class Sorting
             [ArbitraryDimensionArrayImport] IList list,
             [ArbitraryDimensionArrayImport] IList keys)
         {
-            var sortedPairs = list.Cast<object>()
-                    .Zip(keys.Cast<IComparable>(), (item, key) => new { item, key })
-                    .OrderBy(x => x.key)
-                    .ToList();
+            var pairs = list.Cast<object>()
+                    .Zip(keys.Cast<object>(), (item, key) => new { item, key });
+
+            var numberKeyPairs = pairs.Where(pair => pair.key is double || pair.key is int);
+            var stringKeyPairs = pairs.Except(numberKeyPairs);
+
+            // Sort.
+            numberKeyPairs = numberKeyPairs.OrderBy(pair => Convert.ToInt64(pair.key));
```

Comment:
![screen shot 2015-10-07 at 11 32 36 am](https://cloud.githubusercontent.com/assets/508936/10342472/307bc35a-6ce7-11e5-8675-7101b2d55a7e.png)

see this counter example